### PR TITLE
Avoid code repetition when setting compilers

### PR DESCRIPTION
**Pull request by bast**
Compilers are now set in `.scripts/menu.yml` but can be overruled in local `menu.yml`. IMO great relief when creating new scripts and tweaking CI setups.

### CODE REVIEW

File: .scripts/menu.yml [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
 travis-linux:
   definitions:
-    - CMAKE_CXX_COMPILER: 'g++'
     - CMAKE_C_COMPILER: 'gcc'
+    - CMAKE_CXX_COMPILER: 'g++-7'
     - CMAKE_Fortran_COMPILER: 'gfortran'
 
 travis-osx:
   definitions:
+    - CMAKE_C_COMPILER: 'gcc'
```
Comment (robertodr) [inline .scripts/menu.yml]: Did you mean `clang` here?

File: Chapter03/recipe-07/cxx-example/menu.yml [hunk 2]
```diff
@@ -1,27 +1,22 @@
...
 travis-linux:
   definitions:
-    - CMAKE_CXX_COMPILER: 'g++'
     - Eigen3_DIR: '$HOME/Deps/eigen/share/eigen3/cmake/'
 
 travis-osx:
   definitions:
-    - CMAKE_CXX_COMPILER: 'clang++'
     - Eigen3_DIR: '$HOME/Deps/eigen/share/eigen3/cmake/'
   failing_generators:
```
Comment (robertodr) [inline Chapter03/recipe-07/cxx-example/menu.yml]: Why are these expected to fail? I can't remember...

File: .scripts/menu.yml (hunk 1 above)
Comment (bast) [reply .scripts/menu.yml]: Oops. Yes. I will fix that.

File: Chapter03/recipe-07/cxx-example/menu.yml (hunk 2 above)
Comment (bast) [reply Chapter03/recipe-07/cxx-example/menu.yml]: I don't know. Should I reactivate and we see what happens?

Comment (robertodr) [reply Chapter03/recipe-07/cxx-example/menu.yml]: Can't get redder than it already is :smile: I can take over OS X CI business once this one is merged.

Comment (bast) [reply Chapter03/recipe-07/cxx-example/menu.yml]: If we deactivate stuff we should leave comments there in future. I would reactivate everything except what we definitely expect to fail and add some comments.

Comment (robertodr) [reply Chapter03/recipe-07/cxx-example/menu.yml]: Agreed. Want to do it with this PR?

**Comment by bast** [comment]
Ready from my point of view. Please merge when you think it is ready.

**Comment by robertodr** [comment]
I'll wait a moment for Linux CI to run

**Comment by robertodr** [comment]
Pulling this in. I will look at fixing CI ASAP.

**Pull request merged**